### PR TITLE
make thread-num for nydusd should be configurable

### PR DESCRIPTION
**Pull request by liubin**
Use a user configurable for thread-num than a fixed value.

Fixes: #19

Signed-off-by: bin liu <liubin0329@gmail.com>

### CODE REVIEW

File: snapshot/snapshot.go [hunk 1]
```diff
@@ -113,6 +113,8 @@ func NewSnapshotter(ctx context.Context, cfg *config.Config) (snapshots.Snapshot
 		nydus.WithLogLevel(cfg.LogLevel),
 		nydus.WithLogDir(cfg.LogDir),
 		nydus.WithLogToStdout(cfg.LogToStdout),
+		nydus.WithLogToStdout(cfg.LogToStdout),
```
Comment (changweige) [inline snapshot/snapshot.go:116]: Duplicated code line?

File: cmd/containerd-nydus-grpc/pkg/command/flags.go [hunk 2]
```diff
@@ -182,6 +183,11 @@ func buildFlags(args *Args) []cli.Flag {
 			Usage:       "whether to disable nydus-overlayfs to mount",
 			Destination: &args.EnableNydusOverlayFS,
 		},
+		&cli.IntFlag{
+			Name:        "nydusd-thread-num",
+			Usage:       "Nydusd daemon thread-num, default will be set to the number of CPUs",
```
Comment (changweige) [inline cmd/containerd-nydus-grpc/pkg/command/flags.go:188]: I am not very sure if make the default threads number to the number of CPUs is a suitable policy.
Consider that a server equipped with many CPU sockets and cores, dozens of threads will be spawned. But most of them are idle.

Comment (liubin) [reply cmd/containerd-nydus-grpc/pkg/command/flags.go:188]: It seems nydusd is using a default of `1`, so if we don't want to set a default value from snapsohtter, we can just don't pass this argument to nydusd if the user doesn't set it.

Comment (changweige) [reply cmd/containerd-nydus-grpc/pkg/command/flags.go:188]: Sounds reasonable

Comment (liubin) [reply cmd/containerd-nydus-grpc/pkg/command/flags.go:188]: fixed.

File: snapshot/snapshot.go (hunk 1 above)
Comment (liubin) [reply snapshot/snapshot.go]: Deleted, my keyboard is broken...

**Review by changweige — APPROVED** [review]
Thanks

**Pull request merged**